### PR TITLE
Enhance/vite

### PR DESCRIPTION
**Pull request by vgkids**
**Before**
App is compiled by Gulp + Webpacker

**After**
App is compiled using Vite.

Note: Building apps and running Lamda-backed dashboard panels still requires the Express server, but these features are now default-off and enabled with env vars. This allows the app to be served as static files in environments where these features are not needed.

To test features related to app building and running, you need to manually build the app and start the server after making any changes:
`npm run build && npm run start`

If you don't need app building & running, you can use modern Vite tooling:

Start the development server:
`npm run dev`

BEFORE PROMOTE

- [X] Set these values on production

```shell
VITE_TRIVIAL_API_URL=https://trivial-api-production.herokuapp.com
VITE_ENABLE_SAVE_CREDENTIALS = true
VITE_ENABLE_BUILD_APPS = true
VITE_ENABLE_WEBHOOK_APP_TRIGGER = true
```

RELEASE BLOCKERS
- [x] env var name changes; is there a better way?
- [x] build the action registry. gulp was doing this, but is gone.

### CODE REVIEW

**Comment by elmerv** [comment]
I was able to get it working on my end after changing environment variables and ports, since Vue app now runs on 5173. 

Features tested:
- Logging In ✅ 
- Signing up ✅ 
- Sending/Accepting an invitation ✅ 
- Creating an app ❌ 
- Resetting Password ✅ 

## Creating App Error
Trivial-UI loads and all the functionality works except when it comes to creating new apps. When you create a dashboard, there is an error in trivial-core that occurs when converting fetched initial_manifest to json. 

### Error in Console:
[fetchJSON] could not parse JSON response: SyntaxError: Unexpected token '<', "<html>
<he"... is not valid JSON


_This causes this error on UI to occur:_
Screen Shot 2024-03-08 at 12 14 09 PM

 Despite the error, the app does get made and shows up on dashboard table. Sub-apps/charts encounter the same error. Sub-apps are also made (they show up via url, ex: headline, but don't appear in dashboard on refresh. 

## Theme Error

The last bug I encountered had to do with the theme of trivial-ui staying after you log out. If you put the trivial-ui theme to dark before logging out, it stays as dark.

Screen Shot 2024-03-08 at 12 50 05 PM


## Unrelated error: 
notifications is not imported in createPanel.js, this causes an unimported Notifications error when creating a dashboard error.

File: .env.example [hunk 1]
```diff
@@ -5,7 +5,7 @@ PORT=3000
 # Optional settings below.
 #
 # Your trivial-ui-backend instance:
-# TRIVIAL_URL=https://trivial-api-staging.herokuapp.com
+# VUE_APP_TRIVIAL_API_URL=https://trivial-api-staging.herokuapp.com
```
Comment (elmerv) [inline .env.example:8]: Not sure what it should be named, but the variable in index.js in store is set to TRIVIAL_API_URL.

**Comment by vgkids** [comment]
@elmerv Please run through this again. We'll go live with the theme error you mention, but the rest should be resolved.

Note: This branch is live at https://www.staging.trivialapps.io/ so I could work through prod-like build issues before merging.

You'll need to run `npm install`, and you may need to sign out and back in again to get apps building correctly.

**Comment by elmerv** [comment]
### Testing Report:

Testing environment:
- all `VITE_` vars were ommitted when testing locally
-  TRIVIAL-API (main) and TRIVIAL-UI (latest enhance/vite) locally

Results:
| Feature             | Tested on Staging |
|:-------------------:|:-----------------:|
| Sign Up             |        ✅         |
| Creating Apps       |        ✅         |
| App Saves and Runs  |        ✅         |
| Sign In             |        ✅         |
| App Settings        |        ❌         |
| Sign Out            |        ❓         |

| Features Tested on Local | Status |
|--------------------------|--------|
| All features from first comment       |   ✅   |
| App Settings             |   ✅   |
| App Saves and Runs             |   ✅   |
| Sign Out                 |   ❓   |

Errors
---

### App Settings for newly created apps:
- On staging, app setting information excluding transfer apps and manifest is not loading.
   - Step to Reproduce
       1) Create a new dashboard app
       2) Navigate to dashboard advanced settings page
- On local, this error does not occur, app setting information loads. 

Images on trivial-ui staging: 
Screen Shot 2024-03-14 at 12 08 01 PM
Screen Shot 2024-03-14 at 12 08 18 PM

### Signing out provides console.errors _(Possible unrelated error to branch)_ 
- Currently, signing out is providing two console errors. Both of these errors occur locally and on staging:
Screen Shot 2024-03-14 at 12 13 24 PM
- Despite the errors, signing out does work as normal. Navigating to /dashboard page after signing out redirects to sign in page.
@vgkids

**Comment by vgkids** [comment]
@elmerv 
App Settings: staging was a few commits behind. I've rebuilt and it should be working now
Signout error, please create a separate issue and we'll tackle post-merge

**Review by drowninginflowers — APPROVED** [review]
Please confirm the new commit history. All changes approved!

**Pull request merged**